src/gt4py/gtc/python/oir_to_npir.py

```diff
@@ -0,0 +1,104 @@
+from typing import Dict, List, Optional
+
+from dataclasses import dataclass, field
+from eve.visitors import NodeTranslator
+
+from .. import common, oir
+from . import npir
+
+
+class OirToNpir(NodeTranslator):
+    """Lower from optimizable IR (OIR) to numpy IR (NPIR)."""
+
+    @dataclass
+    class Context:
+        """Context for a HorizontalExecution."""
+
+        domain_padding: Dict[str, List] = field(
+            default_factory=lambda: {"lower": [0, 0, 0], "upper": [0, 0, 0]}
+        )
+        parallel_k: Optional[bool] = None
```

Comment:
I wouldn't make `parallel_k` part of the context object, because information flows only to children. With this approach `parallel_k` will be set to a value outside its scope. I think here the better pattern is just pass it down via kwargs.

Reply:
`parallel_k` is never set outside the scope as long as `set_parallel_k()` is used. I will make the attribute hidden so it is more obvious it shouldn't be set directly.

Reply:
What I mean is, after `set_parallel_k()` is called it will have a value, even outside of the scope where it makes sense. Or do I misunderstand the mechanism?